### PR TITLE
[bedcov] New options (-g, -G, -d)

### PR DESCRIPTION
**Pull request by valeriuo**
`samtools bedcov` can now filter reads in and out, depending on the attached flag, using the new options `-g` and `-G` (similar to `depth`).

Option `-d <val>` allows the user to display an extra column for each file, containing the number of bases with a coverage above and including `<val>`. The extra columns are displayed after all the coverage columns and only if this option is used.

```
-g <flags>          include reads that have any of the specified flags set [0]
-G <flags>          filter out reads that have any of the specified flags set
                           [UNMAP, SECONDARY, QCFAIL, DUP]
```

### CODE REVIEW

File: bedcov.c [hunk 1]
```diff
@@ -188,13 +217,21 @@ int main_bedcov(int argc, char *argv[])
                             if (pi->is_del || pi->is_refskip) ++m;
                         }
                     }
-                    cnt[i] += n_plp[i] - m;
+                    int pd = n_plp[i] - m;
+                    cnt[i] += pd;
+                    if (min_depth >= 0 && pd > min_depth) pcov[i]++;
```
Comment (daviesrob) [inline bedcov.c]: Should this be `pd >= min_depth`?  Help message says "Number of bases with coverage above and including this value".

File: bedcov.c [hunk 2]
```diff
@@ -96,7 +116,12 @@ int main_bedcov(int argc, char *argv[])
         fprintf(stderr, "Options:\n");
         fprintf(stderr, "      -Q <int>            mapping quality threshold [0]\n");
         fprintf(stderr, "      -X                  use customized index files\n");
+        fprintf(stderr, "      -g <flags>          include reads that have any of the specified flags set [0]\n");
+        fprintf(stderr, "      -G <flags>          filter out reads that have any of the specified flags set"
```
[the remark this answers is not in the document]
Comment (valeriuo) [reply bedcov.c]: Yes. The use of these options is slightly convoluted, due to the default flag. I will updated the help message.

File: bedcov.c [hunk 3]
```diff
@@ -177,7 +205,8 @@ int main_bedcov(int argc, char *argv[])
         }
         mplp = bam_mplp_init(n, read_bam, (void**)aux);
         bam_mplp_set_maxcnt(mplp, 64000);
```
[the remark this answers is not in the document]
Comment (valeriuo) [reply bedcov.c]: Actually, it does make sense to increase the mpileup depth limit, when using the new option, even if the use case might not exist today.

File: bedcov.c (hunk 1 above)
Comment (valeriuo) [reply bedcov.c]: I see that mpileup returns a reference skip, by default. I might have been misled by my test data.

File: bedcov.c [hunk 4]
```diff
@@ -96,7 +116,12 @@ int main_bedcov(int argc, char *argv[])
         fprintf(stderr, "Options:\n");
         fprintf(stderr, "      -Q <int>            mapping quality threshold [0]\n");
         fprintf(stderr, "      -X                  use customized index files\n");
+        fprintf(stderr, "      -g <flags>          include reads that have any of the specified flags set [0]\n");
+        fprintf(stderr, "      -G <flags>          filter out reads that have any of the specified flags set"
+                        "                          [UNMAP,SECONDARY,QCFAIL,DUP]\n");
         fprintf(stderr, "      -j                  do not include deletions (D) and ref skips (N) in bedcov computation\n");
+        fprintf(stderr, "      -d <int>            depth threshold. Number of bases with coverage above and"
```
[the remark this answers is not in the document]
Comment (valeriuo) [reply bedcov.c]: > Also, is there potential for confusion with the depth -d option, which does something subtly different?

We can go with something else, `-t` or `-m`, but I wouldn't be surprised if the same is used for a different purpose by another samtools command.

**Pull request merged**